components/camel-mail/src/test/java/org/apache/camel/component/mail/MailDisconnectTest.java

```diff
@@ -39,14 +43,17 @@ public void testDisconnect() throws Exception {
         template.sendBodyAndHeader(jones.uriPrefix(Protocol.smtp), "A Bla bla", "Subject", "Hello A");
         template.sendBodyAndHeader(jones.uriPrefix(Protocol.smtp), "B Bla bla", "Subject", "Hello B");
 
-        Thread.sleep(500);
-        template.sendBodyAndHeader(jones.uriPrefix(Protocol.smtp), "C Bla bla", "Subject", "Hello C");
+        Awaitility.await().pollDelay(500, TimeUnit.MILLISECONDS).untilAsserted(() -> Assertions
+                .assertDoesNotThrow(
+                        () -> template.sendBodyAndHeader(jones.uriPrefix(Protocol.smtp), "C Bla bla", "Subject", "Hello C")));
 
-        Thread.sleep(500);
-        template.sendBodyAndHeader(jones.uriPrefix(Protocol.smtp), "D Bla bla", "Subject", "Hello D");
+        Awaitility.await().pollDelay(500, TimeUnit.MILLISECONDS).untilAsserted(() -> Assertions
+                .assertDoesNotThrow(
+                        () -> template.sendBodyAndHeader(jones.uriPrefix(Protocol.smtp), "D Bla bla", "Subject", "Hello D")));
 
-        Thread.sleep(500);
-        template.sendBodyAndHeader(jones.uriPrefix(Protocol.smtp), "E Bla bla", "Subject", "Hello E");
+        Awaitility.await().pollDelay(500, TimeUnit.MILLISECONDS).untilAsserted(() -> Assertions
+                .assertDoesNotThrow(
+                        () -> template.sendBodyAndHeader(jones.uriPrefix(Protocol.smtp), "E Bla bla", "Subject", "Hello E")));
```

Comment:
@davsclaus this is also not correct, right?

Reply:
Yes this is wrong as well, as you can risk having it send the message more than once as the code is called again

Reply:
Okay, thanks!

Reply:
@davsclaus i have updated the code. Can you please review it?
Thanks!